src/base/static/components/input-form/index.js

```diff
@@ -314,73 +304,90 @@ class InputForm extends Component {
         attrs[field.name] = extractEmbeddedImages(attrs[field.name]);
       });
 
-    this.attachments.forEach(attachment => {
-      model.attachmentCollection.add(attachment);
-    });
-
     // Fire pre-save hook.
     // The pre-save hook allows flavors to attach arbitrary data to the attrs
     // object before submission to the database.
     if (this.props.customHooks && this.props.customHooks.preSave) {
       attrs = hooks[this.props.customHooks.preSave](attrs);
     }
 
-    model.save(attrs, {
-      success: response => {
-        Util.log("USER", "new-place", "successfully-add-place");
+    const response = await mapseedApiClient.place.create({
+      datasetUrl: this.props.datasetUrl,
+      placeData: attrs,
+      datasetSlug: this.props.datasetSlug,
+      clientSlug: this.props.datasetClientSlugSelector(this.props.datasetSlug),
+    });
+
+    if (response) {
```

Comment:
In general, interacting with the API is a two-step process: first we send the network request, and then, if it's successful, we update the appropriate duck. The ducks are always update outside the API client itself, but I wonder if the client should also be responsible for keeping ducks in sync with the result of network requests. This would more closely duplicate what the Backbone models' responsibility, but it would mean that the client would need access to the Redux store.

What do you think?

Reply:
I think we should keep our networking and serialization layer separate from our redux store, so I think this is reasonable.

But is having the api client separate from the redux store causing issues?

Although I suppose this might be more readable if we wrote it like this:

```js
if (!response) {
  alert("Oh dear. It looks like that didn't save. Please try again.");
  Util.log("USER", "place", "fail-to-create-place");
  return
}

// do stuff with the response...
```

We should also consider having MapseedApiClient throw an error if the response is not a 200 and using a try/catch pattern. I'm not sure what the best practice is, or if there even is one.

Reply:
Ok, I'll leave the two separated. I don't think it's causing an issue, it just occurred to me that it might be cleaner for the API to manage both network requests and syncing local state (a la Backbone).

We do have some HTTP error handling in place in the API client-- all requests get passed through the `response` handler and will throw an error if the HTTP response code is not in the 200 range.